02-microarray/pathway-analysis_microarray_03_gsea.Rmd

```diff
@@ -333,15 +336,169 @@ Looks like we were able to successfully get rid of the duplicate gene identifier
 
 ## Perform gene set enrichment analysis (GSEA)
 
-_Addressed in upcoming PR_
+The goal of GSEA is to determine whether the members of a set of genes are randomly distributed throughout a pre-ranked gene list or primarily found at the top or bottom of the list meaning that they are highly- or lowly-ranked.
+As mentioned earlier, hallmark gene sets summarize and represent specific well-defined biological states or processes and display coherent expression of genes so we would expect the latter distribution. [@clusterProfiler-book].
+
+### Determine our pre-ranked genes list
+
+The `GSEA()` function takes a pre-ranked and sorted named vector of statistics, where the names in the vector are gene identifiers.
+This is step 1 -- create a gene list with statistics that GSEA will rank by.
+
+```{r}
+# Let's create a named vector ranked based on the t-statistic values
+t_vector <- filtered_dge_mapped_df$t
+names(t_vector) <- filtered_dge_mapped_df$entrez_id
+
+# We need to sort the t-statistic values in descending order here
+t_vector <- sort(t_vector, decreasing = TRUE)
+```
+
+Let's preview our pre-ranked named vector.
+
+```{r}
+# Look at first entries of the ranked t-statistic vector
+head(t_vector)
+```
+
+### Run GSEA using the `GSEA()` function
+
+Genes are ranked from most highly positive to most highly negative and weighted according to their gene-level statistic.
+The enrichment score (ES) is a pathway-level statistic, calculated using our gene-level statistics.
+Normalized enrichment scores (NES) are enrichment scores that are scaled to make gene sets that contain different number of genes comparable [@Subramanian2005; @Korotkevich2019].
+
+We can use the `GSEA()` function to perform GSEA with any generic set of gene sets, but there are several functions for using specific, commonly used gene sets (e.g., `gseKEGG()`).
+
+```{r}
+gsea_results <- GSEA(
+  geneList = t_vector, # ordered ranked gene list
+  minGSSize = 25, # minimum gene set size
+  maxGSSize = 500, # maximum gene set set
+  pvalueCutoff = 0.05, # p value cutoff
+  eps = 0, # boundary for calculating the p value
+  seed = TRUE, # set seed to make results reproducible
+  pAdjustMethod = "BH", # Benjamini-Hochberg correction
+  TERM2GENE = dplyr::select(
+    dr_hallmark_df,
+    gs_name,
+    entrez_gene
+  )
+)
+```
+
+Let's take a look at the results.
+
+```{r}
+# We can access the results from our gseaResult object using `@result`
+head(gsea_results@result)
+```
+
+Significance is assessed by generating a null distribution by sampling random gene sets of the same size and an FDR (false discovery rate) value is calculated to account for multiple hypothesis testing.
```

Comment:
I _believe_ that this implementation permutes the gene labels and uses the same gene set? It's basically accomplishing the same thing – you get an distribution of ES values for a gene sets that are the same size as the _real_ gene set but the relationship between the gene set and the true ranking is "broken."  I would describe it the same way that the docs or book for clusterProfiler do. (I think this may come directly from training material and I made it a point to talk about there would be two different ways to accomplish the goal / I just wanted the point to land using whatever mental model worked best for someone.) I may be overthinking it. I can't tell.

Reply:
Gotcha! I'll revisit the docs and be sure to describe this section in a way which is more closely aligned with what's in the docs relevant to the package version we are using.